hedera-node/src/main/java/com/hedera/services/txns/token/TokenWipeTransitionLogic.java

```diff
@@ -46,34 +51,57 @@
  */
 public class TokenWipeTransitionLogic implements TransitionLogic {
 	private static final Logger log = LogManager.getLogger(TokenWipeTransitionLogic.class);
-
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
-
-	private final TokenStore store;
 	private final TransactionContext txnCtx;
+	private final TypedTokenStore tokenStore;
+	private final AccountStore accountStore;
+	private final OptionValidator validator;
+	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
 	public TokenWipeTransitionLogic(
-			TokenStore store,
+			final OptionValidator validator,
+			TypedTokenStore tokenStore,
+			AccountStore accountStore,
 			TransactionContext txnCtx
 	) {
-		this.store = store;
 		this.txnCtx = txnCtx;
+		this.tokenStore = tokenStore;
+		this.accountStore = accountStore;
+		this.validator = validator;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			var op = txnCtx.accessor().getTxn().getTokenWipe();
-			var outcome = store.wipe(op.getAccount(), store.resolve(op.getToken()), op.getAmount(),false);
-			var id = store.resolve(op.getToken());
-			txnCtx.setStatus((outcome == OK) ? SUCCESS : outcome);
-			if(outcome == OK) {
-				txnCtx.setNewTotalSupply(store.get(id).totalSupply());
-			}
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
+		/* obtain grpc objects */
+		var op = txnCtx.accessor().getTxn().getTokenWipe();
+		final var grpcTokenId = op.getToken();
+		final var grpcAccountId = op.getAccount();
+
+		/* translate to actual ids */
+		final var targetTokenId = new Id(grpcTokenId.getShardNum(), grpcTokenId.getRealmNum(), grpcTokenId.getTokenNum());
+		final var targetAccountId = new Id(grpcAccountId.getShardNum(), grpcAccountId.getRealmNum(), grpcAccountId.getAccountNum());
+
+		/* load model objects */
+		final var token = tokenStore.loadToken(targetTokenId);
+		final var account = accountStore.loadAccount(targetAccountId);
+
+		/* load relations on the model objects */
+		final var treasuryRel = tokenStore.loadTokenRelationship(token, token.getTreasury());
```

Comment:
I think that you won't event need to load the `treasuryRel` if the changes from the comments are applied. The only place you need the treasury is to verify that the `account` being wiped is not the `treasury` and you can do that using `token.getTreasury()`